x-pack/legacy/plugins/infra/server/lib/snapshot/metric_aggregation_creators/network_traffic.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraNodeType } from '../../../graphql/types';
+import { rate } from './rate';
```

Comment:
It doesn't look like anything still imports `rate` besides `network_traffic` now, unless I'm missing something? Why add a new file instead of just adding all this code to `rate`?

Reply:
Good question. You are right, this is the only place importing `rate` now. I didn't want to add this code to `rate` as it's specific functionality that wouldn't extend well to other metrics. Whereas the `rate` implementation is generic, and could be used for calculating the `rate` of many other things (or Kubernetes here). Usually, I'm of the opinion to "kill it if it's not used", but as we have AWS metrics coming on the horizon, I think a generic `rate` implementation may well be useful again soon.